Automattic/jetpack
# Jetpack Docker: More flexible implementation

The idea of this PR was to allow launching multiple containers using the same(-ish) configuration. It is based on the special `COMPOSE_PROJECT_NAME` ENV variable that could be used to dynamically change compose project name, which basically allows launching multiple containers using different project names. For now, multiple containers are only supported for non-default(`dev`) type - `e2e` 

The motivation behind this is to be able to run E2E tests in parallel, which requires creating multiple containers per run.

To support all the above, this PR includes the re-write of the existing `jetpack docker` CLI commands

There are 2 types of configurations available:
- `dev`, which is a default one. It should be backward compatible with the previous implementation. Runs WP container on port 80.
- `e2e` - barebones configuration, without `maildev`, `sftp` and `phpmyadmin` containers. Also, it does not install additional plugins on container creation. Default WP port - 8889, but can be changed via cli option.

Note: I have not converted `ngrok` setup, since I'm not sure if anyone is still using it.

#### Jetpack product discussion


1156386383419466-as-1199944020854263/f

#### Does this pull request change what data or activity we track or use?


Nope
#### Testing instructions:






* Check out the branch
* Test out your default workflows, it should work as expected.
* Launch multiple containers with the following commands (use `-d` to launch containers in detached mode):
	* `jetpack docker --type e2e up` - will launch default `e2e` project on port 8889.
	* `jetpack docker --type e2e --name test1 --port 8888 up` - will launch `test1_e2e` project on port 8888. 
	* make sure both containers accessible via localhost:PORT, and they are indeed different sites
